shop/src/main/java/ro/msg/learning/configuration/FormBasedWebSecurityConfiguration.java

```diff
@@ -0,0 +1,53 @@
+package ro.msg.learning.configuration;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Profile;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.authentication.configurers.provisioning.InMemoryUserDetailsManagerConfigurer;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+import lombok.RequiredArgsConstructor;
+import ro.msg.learning.entity.CustomerCredentials;
+import ro.msg.learning.service.interfaces.CustomerCredentialsService;
+
+@Configuration
+@EnableWebSecurity
+@RequiredArgsConstructor
+@Profile("form-based")
```

Comment:
FYI: a different application.properties file can be used for each profile (e.g. application-dev.properties and application-test.properties) to configure things for different environments